app/javascript/src/Dashboard/components/ApiFilter.jsx

```diff
@@ -10,7 +10,14 @@ import { render } from 'react-dom'
 
 import 'Dashboard/styles/dashboard.scss'
 
-const ApiFilter = ({ apis, displayApis }) => {
+import type { Api } from 'Types'
```

Comment:
In other [PR](https://github.com/3scale/porta/pull/373/files#diff-e320c75b43a806ef22b64482e372b211R12), we agreed to keep types in the same file.
I'm not against having types in a separated file, but we all should agree on a single way to do it

Reply:
Api type should be defined within `Api.jsx` but we don't have that. The problem with Api's types is that it's already being used in many files (ApiFilter, ContextSelector, ActiveMenuTitle) so that it would lead to repetition.

Reply:
Yeah, DRY, I just say we should agree on a way to handle this. It could be having a general `Types` directory, or having corresponding types inside each component directory

Reply:
If a type is used *only* in one file, then define it there in the same file. If it's shared with other components, then yeah, define it in a separated file in order to DRY it up.

Reply:
Well I would disagree with that @didierofrivia If it is private for the file yes then define there but if not probably better to use a directory convention.

Can a type be **private** btw?

Reply:
There's no such concept of public/private classes. One does not need to import types in order to _read_ them, meaning, if I import a class A, I don't need to import its types, my IDE will show me those automatically.
On the other hand, if I want to define a class B that uses some shared properties/classes with A, I would have to import their types from class A or defined them again within class B. In which case we would move those types to a common directory _types_ or whatever.

React-level stuff should be defined within classes, ie: Props, State
Project-level stuff should be defined in the types dir, ie: Api, Service (models)